inbm/telemetry-agent/telemetry/telemetry_handling.py

```diff
@@ -303,7 +303,7 @@ def get_query_related_info(option: str, info: Dict) -> Dict:
                 del info['values'][elem]
     elif option == "version":
         info = {}
-        info['INBMVersionCommit'] = get_friendly_inbm_version_commit()        
+        info['INBMVersionCommit'] = get_friendly_inbm_version_commit()
     elif option == "all":
```

Comment:
**<h2>$\color{#FFB6C1}Enhancement: PerformanceEnhancement$</h2>**
**Suggestion:** Optimize the get_dynamic_telemetry function by reducing the number of calls to system information retrieval functions if they are not needed for the current telemetry collection.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                dynamic_telemetry: Dict[str, Optional[Any]] = {}
                if 'systemCpuPercent' in required_telemetry_fields:
                    dynamic_telemetry['systemCpuPercent'] = get_cpu_percent()
                ...
                if 'friendlyINBMVersionCommit' in required_telemetry_fields:
                    dynamic_telemetry['friendlyINBMVersionCommit'] = get_friendly_inbm_version_commit()
```

Reply:
good suggestion but out of scope